pyproject.toml

```diff
@@ -36,36 +36,33 @@ lines_between_types = 1
 
 [tool.towncrier]
 package = "plasmapy"
+name = "PlasmaPy"
 filename = "CHANGELOG.rst"
 directory = "changelog/"
-issue_format = "`#{issue} <https://github.com/plasmapy/plasmapy/pull/{issue}>`__"
+wrap = true
+title_format = "{name} v{version} ({project_date})"
+issue_format = ":issue:`{issue}`"
 
-[[tool.towncrier.type]]
-directory = "breaking"
+[tool.towncrier.type.breaking]
 name = "Backwards Incompatible Changes"
 showcontent = true
 
-[[tool.towncrier.type]]
-directory = "removal"
-name = "Deprecations and Removals"
+[tool.towncrier.type.bugfix]
+name = "Bug Fixes"
 showcontent = true
 
-[[tool.towncrier.type]]
-directory = "feature"
-name = "Features"
+[tool.towncrier.type.doc]
+name = "Improved Documentation"
 showcontent = true
 
-[[tool.towncrier.type]]
-directory = "bugfix"
-name = "Bug Fixes"
+[tool.towncrier.type.feature]
+name = "Features"
 showcontent = true
 
-[[tool.towncrier.type]]
-directory = "doc"
-name = "Improved Documentation"
+[tool.towncrier.type.removal]
+name = "Deprecations and Removals"
 showcontent = true
 
-[[tool.towncrier.type]]
-directory = "trivial"
+[tool.towncrier.type.trivial]
```

Comment:
I've been running into some weird stuff with this new configuration.  It's just buggy and hard to explain the issues, but in short... This new configuration syntax was [added back in Feb 24, 2022](https://github.com/twisted/towncrier/commit/bda743b945f8cbaed088a9ba8cceacc9d1c98a79), so it's not even available in the current [21.9.0 release (Feb. 10, 2022)](https://pypi.org/project/towncrier/#history).  As such, `sphinx_changelog` (which we use to generate the dev changelog) has not incorporated the current version of `towncrier`.

TL;DR We need to put this back to the old style configuration to avoid `towncrier` bugs.

Reply:
Sigh!